sg_otio/sg_cut_track_writer.py

```diff
@@ -1088,7 +1088,7 @@ def _create_input_media_version(self, input_media, track_name, sg_project, sg_li
             # SG doesn't seem to accept backslashes when creating
             # PublishedFiles with relative paths to local storage
             relative_path = os.path.normpath(relative_path)
-            relative_path = relative_path.replace("\\", "/")
+            relative_path = relative_path.replace("\\", "/").strip("/")
```

Comment:
Hmm, that does not seem right... It seems that something else should be fixed rather than having this here.

Reply:
Well, with the delivery_fw the responsibility of assigning a relative path is of the FormatCopier, and it's a bit smarter.
In the fixes we did for v3, we rely on the deliverer, and we use it to come up with the relative paths and the way to do it is more thorough, instead of the above :

```
if local_storage and input_media.startswith(local_storage["path"]):
            relative_path = input_media[len(local_storage["path"]):]
```

So the right solution would be in this case to try to extract only the logic that comes up with the relative path?

Reply:
What I mean is that since we use normpath, I guess the strip is to remove "/" at the front. So it means that the path can be absolute, keep in mind that this is just a guess. If so, why in the first place can it be an absolute path? What about Windows where you could have a drive letter in front of the path?
If I understood well the intent, the fix should be where we build this relative path, to make sure it is relative.

Reply:
We build it just above, as mentioned, by doing:

`relative_path = input_media[len(local_storage["path"]):]`

In the delivery fw, we don't compare it simply by doing `local_storage and input_media.startswith(local_storage["path"])` and we don't compute the relative path like we do above.

To me, that's the problem. The fix here is because the replace we do keeps the leading "/", but it should never be a path that starts with "/" because we are trying to compute a relative path just above.

So the fix works if we keep this naive approach of path comparison and replacement. If we don't want to keep the naive approach, we need to extract a bit of logic from a couple of places in the delivery fw.

Reply:
Well the fix should then be: `relative_path = relative_path[1:]`, after the path is normalized, or similar, with a comment explaining why we do that instead of removing "/" at the front. Or add a separator at the end of `local_storage["path']` so storage "/foo/bar" will not match "/foo/barmitzta", which seems to be wrongly the case...

Reply:
You are right, we should make sure the local storage path ends with a separator.